gigaword-indexer/src/main/java/edu/isi/vista/gigawordIndexer/IndexGigawordWithElasticSearch.java

```diff
@@ -135,20 +167,40 @@ private static void index(RestHighLevelClient client, Path file, String indexNam
 
     // we batch the documents in groups of 100 so we can get the efficiency gains from batching without
     // making huge requests of unbounded size
-    for (List<Article> articles : partition(ConcatenatedGigawordDocuments.fromGigwordGZippedFile(file), 100)) {
+    Iterable<List<Article>> iterator ;
+    if (format != null && format.equalsIgnoreCase("ltf")) {
+      iterator = partition(LTFDocuments.fromLTFZippedFile(file), BATCH_SIZE);
```

Comment:
This has been moved out of `index` method and up to the caller, so that index method doesn't care about the format. Also it is easier to put it in a try-with-resource block.